### DISCUSSION TITLE
re-usable set of fields as a template?

### DISCUSSION THREAD

**Discussion by eliteproxy7** [directus/directus#6159] [Archive]
I have 20,000 products I want to add as items in a collection, these products are from 100 different vendors.

at first, I just created a collection for each vendor and duplicated the fields .(which ended up causing a lot of issues)

another approach I thought about was to create "template collections" and so many to many relations? would that work? 

here is a quick diagram of what I was thinking.. is this the best approach?

![diagram](https://user-images.githubusercontent.com/5486268/121407687-95457300-c92d-11eb-803d-a581254f306e.jpg)

**Comment by rijkvanzanten** [comment]
I don't really understand your diagram.. Would multiple products share the same general info?

**Reply by eliteproxy7** [reply]
the values for the fields will be unique for each product

**Reply by rijkvanzanten** [reply]
I see. Would a single product be able to have multiple vendors? It sounds to me like you're looking for a single `products` table, with a many-to-many (or many-to-one based depending on my first question) to `vendors`. Next to that, you'll have 2 image fields (which are also `many-to-one`s) for "image field" and "pdf field"

**Reply by eliteproxy7** [reply]
1. product will only have 1 vendor
2. my goal is when I create a new collection for a new vendor, i just want to be able to relate the "general information" collection as a many to one relation to the new collection and enter the field values.. so i don't have to re-create the fields every time

**Reply by rijkvanzanten** [reply]
Gotcha. There's currently no way to duplicate whole groups of fields (only individual ones). That being said, what's preventing you from simply keeping the products in the same collection? That would be a little easier to manage than having 100 duplicate collections with the exact same schema 🤔

**Reply by eliteproxy7** [reply]
i was worried about performance when I have 200/300k products. when we "publish" the products to the website we don't really need them, so I guess "Archiving" would assist with this? performance-wise etc?

**Comment by eliteproxy7** [comment]
i was worried about performance when I have 200/300k products

On Wed, Jun 9, 2021 at 3:04 PM Rijk van Zanten ***@***.***>
wrote:

> Gotcha. There's currently no way to duplicate whole groups of fields (only
> individual ones). That being said, what's preventing you from simply
> keeping the products in the same collection? That would be a little easier
> to manage than having 100 duplicate collections with the exact same schema
> 🤔
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/directus/directus/discussions/6159#discussioncomment-847347>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABJ3NPEZWLSY5JUSR255IRTTR63MFANCNFSM46MR3LTQ>
> .
>


-- 
Hrayr Mirzoyan

**Reply by rijkvanzanten** [reply]
Ohh that should be totally fine. Relational databases are made to hold way more data than that, and Directus itself does filtering etc all in SQL directly 👍🏻

**Reply by eliteproxy7** [reply]
amazing! I greatly appreciate your time
